s3/src/main/scala/geotrellis/store/s3/S3AttributeStore.scala

```diff
@@ -217,7 +217,7 @@ object S3AttributeStore {
   final val SEP = "__"
 
   def apply(bucket: String, root: String, s3Client: => S3Client = S3ClientProducer.get()) =
-    new S3AttributeStore(bucket, root, s3Client)
+    new S3AttributeStore(bucket, Option(root).getOrElse(""), s3Client)
```

Comment:
I think this is fine either way, no change request but question: Is this left over from initial approach or was this a special case that needed to be handled and if so why?

Reply:
The most likely way to get nulls in the first place is addressed by the change to AmazonS3URI, but this was just another spot where a null could exist if a user is passing strings generated by some other Java API. This is a middle ground band-aid versus a more comprehensive update to ensure that all the downstream Layer classes use cleaned bucket/prefix values on the S3AttributeStore, which sometimes doesn't happen now -- each [downstream Layer class holds copies of the bucket/prefix values](https://github.com/locationtech/geotrellis/blob/master/s3-spark/src/main/scala/geotrellis/spark/store/s3/S3LayerWriter.scala#L51-L52) that should theoretically already exist on the S3AttributeStore.